runner/src/error.rs

```diff
@@ -32,6 +33,17 @@ pub enum Error {
     NoChildProcess,
     #[error("A child process is already running")]
     ChildProcessExists,
+    #[error("Failed to terminate child process")]
+    ProcessTerminationFailure,
     #[error("Auto-updater terminated unexpectedly")]
     AutoUpdaterTerminated,
 }
+
+impl<E: Into<Error> + Sized> From<BackoffError<E>> for Error {
+    fn from(e: BackoffError<E>) -> Self {
+        match e {
+            BackoffError::Permanent(err) => err.into(),
+            BackoffError::Transient(err) => err.into(),
+        }
+    }
+}
```

Comment:
Because `retry` returns a nested error type (`backoff::Error<E>`, where `E` is the error returned by the closure). I couldn't figure out a way to add this to the `Error` enum in the runner, since it would be a self-referencing type. I pushed a commit to make the retry calls a bit cleaner and you can now see `map_err` being used instead of the `?`.